model/connection.js

```diff
@@ -101,6 +209,95 @@ class Connection {
         return true;
     }
 
+    /**
+     * Validate the Connection Feature.
+     *
+     * Name, id, connection and layer follow Validation's string rules.
+     * objects.
+     *
+     * @since 1.0.0
+     *
+     * @see validateSinkSourcePoints
+     *
+     * @returns {boolean}
+     */
+    validateFeature() {
+        let valid = super.validate();
+
+        if (!this.connection || !this.connection.validate()) {
+            valid = false;
+            console.log('Connection Feature: Field "connection" is invalid.');
+        }
+
+        valid = this.validateFeatureLayer() ? valid : false;
+        valid = this.validateFeatureConnSegments() ? valid : false;
+
+        if (this.connectionType !== Validation.DEFAULT_CON_TYPE) {
+            valid = false;
+            console.log('Connection Feature: Field "connectionType" is invalid. It must be "'
+                    + Validation.DEFAULT_CON_TYPE + '", but it is "' + this.connectionType + '".');
+        }
+
+        return valid;
+    }
+
+    /**
+     * Validate the Connection Feature's layer.
+     *
+     * The layer must be a valid string. The Connection Feature reference must
+     * be valid. The layer must match the referenced Connection Feature's
+     * layer.
+     *
+     * @since 1.0.0
+     *
+     * @returns {boolean}
+     */
+    validateFeatureLayer() {
+        if (Validation.testStringValue(this.layer, 'layer', 'Connection Feature')) {
+            if (!this.connection) {
+                console.log('Connection Feature: Field "connection" is invalid.');
+                return false;
+            }
+
+            if (this.connection.layer !== this.layer) {
+                console.log('Connection Feature: Field "layer" (' + this.layer + 'does not match the layer in the' +
+                        ' referenced abstract Connection (' + this.connection.layer + ').');
+                return false;
+            }
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate the array of ConnectionSegment objects.
+     *
+     * The length of the array must not be zero. Each ConnectionSegment object
+     * in the array must be valid.
+     *
+     * @since 1.0.0
+     *
+     * @returns {boolean}
+     */
+    validateFeatureConnSegments() {
+        let valid = true;
+
+        if (this.connSegments.length === 0) {
+            return false;
+        }
```

Comment:
Should you have a console.log here stating the issue and setting valid? It seems like the only one that doesn't follow that sequence

Reply:
You're right. It's been added.